### PR TITLE
Feat/tracking platform wallet api flow initiated broadcast success

### PR DESCRIPTION
**Pull request by mitchellv-ledger**
### 📝 Description

Adds flowInitiatedFrom to broadcast success operation tracking for mobile and desktop.
 
### ❓ Context

- **Impacted projects**: `LIVE-8883` 
- **Linked resource(s)**: [] 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-mobile/src/components/Web3AppWebview/helpers.ts [hunk 1]
```diff
@@ -40,6 +41,25 @@ export function useWebView(
   ref: React.ForwardedRef<WebviewAPI>,
   onStateChange: WebviewProps["onStateChange"],
 ) {
+  const tracking = useMemo(
```
Comment (beths-ledger) [inline apps/ledger-live-mobile/src/components/Web3AppWebview/helpers.ts:44]: Does this compute something? Or should it be a useCallback if it's a function?

File: apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx [hunk 2]
```diff
@@ -46,8 +46,20 @@ import { RootNavigationComposite, StackNavigatorNavigation } from "../RootNaviga
 import { BaseNavigatorStackParamList } from "../RootNavigator/types/BaseNavigator";
 import { WebviewAPI, WebviewProps } from "./types";
 import { useWebviewState } from "./helpers";
-
-const tracking = trackingWrapper(track);
+import { currentRouteNameRef } from "../../analytics/screenRefs";
+
+const tracking = (isNativeSwap?: boolean) =>
```
Comment (beths-ledger) [inline apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx:51]:
```suggestion
const trackWithNativeFlowOrigin = (isNativeSwap?: boolean) =>
```

File: apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx [hunk 3]
```diff
@@ -58,6 +70,8 @@ function renderLoading() {
 }
 export const PlatformAPIWebview = forwardRef<WebviewAPI, WebviewProps>(
   ({ manifest, inputs = {}, onStateChange }, ref) => {
+    const isNativeSwap = useMemo(() => currentRouteNameRef.current === "Swap", []);
```
Comment (beths-ledger) [inline apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx:73]:
```suggestion
    const tracking =  trackWithNativeOrigin(currentRouteNameRef.current === "Swap")
```

We can close over the ref here, once. If you instead declare `tracking` here to the function, it closes over the ref, and then the rest of this file can remain the same, and look cleaner.  i.e. `tracking().platformRequestAccountRequested(manifest)` can go back to `tracking.platformRequestAccountRequested(manifest)`

**Review by beths-ledger — COMMENTED** [review]
Left a comment, I think the use of tracking() as a curried function can be improved - let me know what you think

File: apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx (hunk 3 above)
Comment (beths-ledger) [reply apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx:73]: Here is a test demonstrating what I mean:

```jest
function appManifestFixture(name = "live-app"): AppManifest {
  return {
    id: "12",
    name,
    url: "https://www.ledger.fr",
    homepageUrl: "https://www.ledger.fr",
    platforms: ["ios", "android", "desktop"],
    apiVersion: "1.0.0",
    manifestVersion: "1.0.0",
    branch: "debug",
    categories: ["test"],
    currencies: "*",
    content: {
      shortDescription: {
        en: "test",
      },
      description: {
        en: "test",
      },
    },
    permissions: [],
    domains: [],
    visibility: "complete",
  };
}

describe("tracking() with flowInitiatedFrom ", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });
  const mockedTrack = jest.fn();
  const manifest = appManifestFixture();
  const trackWithNativeOrigin = (isNativeSwap?: boolean) =>
    trackingWrapper(
      (
        eventName: string,
        properties?: Record<string, unknown> | null,
        mandatory?: boolean | null,
      ) =>
        mockedTrack(
          eventName,
          {
            ...properties,
            flowInitiatedFrom: isNativeSwap ? "Native" : "Discover",
          },
          mandatory,
        ),
    );

  it("adds Discover to the tracking props by default", () => {
    // When
    const tracking = trackWithNativeOrigin();
    tracking.requestAccountRequested(manifest);

    // Then
    expect(mockedTrack).toHaveBeenCalledTimes(1);
    expect(mockedTrack).toHaveBeenCalledWith(
      "WalletAPI RequestAccount",
      {
        walletAPI: manifest.name,
        flowInitiatedFrom: "Discover",
      },
      null,
    );
  });

  it("adds Native when isSwap is true", () => {
    // When
    const tracking = trackWithNativeOrigin(true);
    tracking.requestAccountRequested(manifest);
    // Then
    expect(mockedTrack).toHaveBeenCalledTimes(1);
    expect(mockedTrack).toHaveBeenCalledWith(
      "WalletAPI RequestAccount",
      {
        walletAPI: manifest.name,
        flowInitiatedFrom: "Native",
      },
      null,
    );
  });
});


```

Comment (mitchellv-ledger) [reply apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx:73]: Unfortunately this was done this way for a reason.  The component rerenders causing the ref to change and losing the original value we needed so has to be memoised

File: apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx (hunk 2 above)
Comment (sarneijim) [reply apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx:51]: native swap is not in a webview.

Comment (beths-ledger) [reply apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx:51]: In that case it's just `trackWithDiscoverOrigin`

File: apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx (hunk 3 above)
Comment (beths-ledger) [reply apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx:73]: that's weird, refs should not be affected by re-renders! But you can still close over the memoized value, `const tracking = trackWithNativeOrigin(isNativeSwap);`. 

Or just trackWithNativeOrigin(false), in this case, potentially, if @sarneijim Sara is saying that it's never native origin in this webview ..?

Comment (beths-ledger) [reply apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx:73]: (I don't know what Native vs Discover origin means)

File: apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx (hunk 2 above)
Comment (beths-ledger) [reply apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx:51]: Ok I am looking at the Slack thread now

**Review by beths-ledger — APPROVED** [review]
I don't fully understand the refs situation going on here but trust that you do!

**Review by sarneijim — APPROVED** [review]

**Pull request merged**